app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,36 @@
+class QueueItemsController < ApplicationController
+  before_action :require_sign_in
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.delete
+    current_user.queue_items.each do |queue_item|
+      queue_item.update(position: queue_item.position - 1)
```

Comment:
I think you might right into some issues with this in the long run...because the position is going to run into negatives. Watch out for how Kevin handles it in the solution.